src/test/java/itest/SnapshotIT.java

```diff
@@ -0,0 +1,305 @@
+
+            MultiMap form = MultiMap.caseInsensitiveMultiMap();
+            form.add("recordingName", TEST_RECORDING_NAME);
+            form.add("duration", "5");
+            form.add("events", "template=ALL");
+
+            webClient
+                    .post(String.format("%s/recordings", V1_TARGET_REQ_URL))
+                    .sendForm(
+                            form,
+                            ar -> {
+                                if (assertRequestStatus(ar, recordResponse)) {
+                                    MatcherAssert.assertThat(
+                                            ar.result().statusCode(), Matchers.equalTo(201));
+                                    recordResponse.complete(null);
+                                }
+                            });
+
+            recordResponse.get();
+
+            // Create a snapshot recording of all events at that time
+            CompletableFuture<String> snapshotResponse = new CompletableFuture<>();
+            webClient
+                    .post(String.format("%s/snapshot", V2_TARGET_REQ_URL))
+                    .send(
+                            ar -> {
+                                if (assertRequestStatus(ar, snapshotResponse)) {
+                                    MatcherAssert.assertThat(
+                                            ar.result().statusCode(), Matchers.equalTo(201));
+                                    MatcherAssert.assertThat(
+                                            ar.result()
+                                                    .getHeader(HttpHeaders.CONTENT_TYPE.toString()),
+                                            Matchers.equalTo(HttpMimeType.PLAINTEXT.mime()));
+                                    MatcherAssert.assertThat(
+                                            ar.result().getHeader("Location"),
+                                            Matchers.equalTo(downloadUrl));
+                                    snapshotResponse.complete(ar.result().bodyAsString());
+                                }
+                            });
+
+            MatcherAssert.assertThat(
+                    snapshotResponse.get(), Matchers.containsString(expectedMetaResponse));
+            MatcherAssert.assertThat(
+                    snapshotResponse.get(), Matchers.containsString(expectedDownloadUrl));
+            MatcherAssert.assertThat(
+                    snapshotResponse.get(), Matchers.containsString(expectedReportUrl));
+            MatcherAssert.assertThat(
+                    snapshotResponse.get(), Matchers.containsString(expectedRecordingId));
+            MatcherAssert.assertThat(
+                    snapshotResponse.get(), Matchers.containsString(expectedRecordingState));
+            MatcherAssert.assertThat(snapshotResponse.get(), Matchers.containsString("startTime"));
+            MatcherAssert.assertThat(
+                    snapshotResponse.get(), Matchers.containsString(expectedRecordingOptions));
```

Comment:
Any ideas to make this section (and the `final String`s above) easier to read?

Reply:
Since what you're really trying to do is verify the structure of a JSON object response, you can remodel that as a Java Map instead of a series of Strings. Example here: https://github.com/cryostatio/cryostat/blob/443928f21124cfba2b94cb1ed0d045834f81c3d6/src/test/java/itest/AutoRulesIT.java#L239